src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -2581,6 +2581,23 @@ public function escapeWhenCastingToString($escape = true)
         return $this;
     }
 
+    protected function getPropertiesForSerialization(): array
+    {
+        if (version_compare(PHP_VERSION, '8.4.0') < 0) {
+            return array_keys(get_object_vars($this));
+        }
+
+        $keys = [];
+
+        foreach ((new ReflectionClass($this))->getProperties() as $property) {
```

Comment:
Since you want to improve the performance, it's advisable to not initialize the `ReflectionClass` within the `foreach` as this will potentially be called multiple times.
```suggestion
        $properties = (new ReflectionClass($this))->getProperties();
        foreach ($properties as $property) {
```

Reply:
Here I rather stick to the same style as in L372.

Reply:
¯\\_ (ツ)_/¯